ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -230,7 +230,14 @@ let symbols_from_cm_file ~filter root_uri (cancel : Fiber.Cancel.t option)
         in
         let loc = Mbrowse.node_loc browse in
         let fname = loc.loc_start.pos_fname in
-        let uri = Uri.of_path (Filename.concat root_uri fname) in
+        let uri =
+          let sourcefile = Filename.concat root_uri fname in
+          let path =
+            if Fpath.exists (Fpath.of_string sourcefile) then sourcefile
+            else build_dir ^ "/" ^ fname
```

Comment:
I think it would be better if we use Fpath for path concatenation, instead of doing it by hand